oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -90,15 +95,23 @@ private ElasticConnection(@NotNull String indexPrefix, @NotNull String scheme, @
         this.apiKeySecret = apiKeySecret;
     }
 
+    /**
+     * Gets the REST High Level Client and instanciates the NEW Elasticsearch client
+     * by sharing the REST High Level Client transport layer
+     * to follow the proposed migration strategy:
+     * https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/7.16/migrate-hlrc.html
+     * It double-checks locking to get good performance and avoid double initialization
+     * @deprecated
+     * @return the old Elasticsearch client
+     */
     public RestHighLevelClient getClient() {
         if (isClosed.get()) {
             throw new IllegalStateException("Already closed");
         }
 
-        // double-checked locking to get good performance and avoid double initialization
-        if (client == null) {
+        if (hlClient == null) {
```

Comment:
If we do use double-checked locking, then we need to use esClient == null here, otherwise there is a possibility that hlClient != null but esClient isn't assigned yet (race condition).

And, at some point in the future, I assume we want to get rid of the hlClient field. So, it's better to not use it where possible.